File: backend/src/application/dtos/Record.dto.js
'use strict';

class CreateRecordDto {
  constructor({ title, description }) {
    if (!title || typeof title !== 'string' || !title.trim()) {
      throw new Error('Title is required and must be a non-empty string');
    }
    this.title = title.trim();
    this.description = typeof description === 'string' ? description.trim() : '';
  }
}

class RecordResponseDto {
  constructor(recordEntity) {
    this.id = recordEntity.id ? String(recordEntity.id) : '';
    this.title = recordEntity.title;
    this.description = recordEntity.description;
    this.createdAt = recordEntity.createdAt instanceof Date
      ? recordEntity.createdAt.toISOString()
      : recordEntity.createdAt;
  }
}

module.exports = {
  CreateRecordDto,
  RecordResponseDto,
};

File: backend/app/application/dtos/record_dto.py
from pydantic import BaseModel, Field
from typing import Optional

class CreateRecordDto(BaseModel):
    title: str = Field(..., min_length=1)
    description: Optional[str] = ""

class RecordResponseDto(BaseModel):
    id: str
    title: str
    description: str
    createdAt: str
